stdlib/Distributed/src/managers.jl

```diff
@@ -184,29 +195,61 @@ function launch_on_machine(manager::SSHManager, machine, cnt, params, launched,
     tval = get(ENV, "JULIA_WORKER_TIMEOUT", "")
 
     # Julia process with passed in command line flag arguments
-    cmds = """
-        cd -- $(shell_escape_posixly(dir))
-        $(isempty(tval) ? "" : "export JULIA_WORKER_TIMEOUT=$(shell_escape_posixly(tval))")
-        $(shell_escape_posixly(exename)) $(shell_escape_posixly(exeflags))"""
+    if shell == :posix
+        # ssh connects to a POSIX shell
+
+        cmdline_cookie = false
+        exeflags = `$exeflags --worker`
+        cmds = """
+            cd -- $(shell_escape_posixly(dir))
+            $(isempty(tval) ? "" : "export JULIA_WORKER_TIMEOUT=$(shell_escape_posixly(tval))")
+            $(shell_escape_posixly(exename)) $(shell_escape_posixly(exeflags))"""
+
+        # shell login (-l) with string command (-c) to launch julia process
+        remotecmd = shell_escape_posixly(`sh -l -c $cmds`)
+
+    elseif shell == :wincmd
+        # ssh connects to Windows cmd.exe
+
+        # Passing the cookie via ssh stdin currently hangs with
+        # Microsoft's Windows port of OpenSSH, so we do it via
+        # --worker on the command line for now, although that may
+        # cause the cookie to become visible to other users (e.g.
+        # using Process Explorer or WMIC).
+        cmdline_cookie = true
+        exeflags = `$exeflags --worker=$(cluster_cookie())`
+
+        any(c -> c == '"', exename) && throw(ArgumentError("invalid exename"))
```

Comment:
should be safe (because of shell-escape below), although true it's not a valid Win32 name.